writer_test.go

```diff
@@ -144,38 +142,16 @@ func TestValidateWriter(t *testing.T) {
 	}
 }
 
-type fakeWriter struct{}
-
-func (f *fakeWriter) messages() chan<- writerMessage {
-	ch := make(chan writerMessage, 1)
-
-	go func() {
-		for {
-			msg := <-ch
-			msg.res <- &writerError{
-				err: errors.New("bad attempt"),
-			}
-		}
-	}()
-
-	return ch
-}
-
-func (f *fakeWriter) close() {
-
-}
-
 func testWriterMaxAttemptsErr(t *testing.T) {
-	const topic = "test-writer-2"
-
+	topic := makeTopic()
 	createTopic(t, topic, 1)
+	defer deleteTopic(t, topic)
+
 	w := newTestWriter(WriterConfig{
+		Brokers:     []string{"localhost:9999"}, // nothing is listening here
 		Topic:       topic,
-		MaxAttempts: 1,
+		MaxAttempts: 3,
```

Comment:
It would be nice to have a bit more thorough testing around how many attempts were actually made, and that writes actually succeed on subsequent attempts if the temporary issue resolves. In the old version, I had started going down the route of putting the `write` method (now `produce`) in an interface so that test code could substitute it with a dumb implementation which counts attempt numbers. Do you have any thoughts on that?

Reply:
also, I think this test will fail if I happen to have a kafka server running on port 9999 😄